### PR TITLE
feat: APP-214 sold out projects

### PR DESCRIPTION
**Pull request by blushi**
## Description

https://regennetwork.atlassian.net/browse/APP-214?atlOrigin=eyJpIjoiYTM4NjU1MjUxZWJmNDU5ZmE4NjM2NjgzOGE3NTY1ZmYiLCJwIjoiaiJ9

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### How to test

1. Sold out prefinance project card: https://deploy-preview-2414--regen-marketplace.netlify.app/projects/prefinance
2. Sold out prefinance project page: https://deploy-preview-2414--regen-marketplace.netlify.app/project/prefinance-project
3. Sold out on chain project card: https://deploy-preview-2414--regen-marketplace.netlify.app/projects/1 and **filter to show only Urban Forest Carbon Credit Class**, 2nd project is sold out
4. Sold out on chain project page: https://deploy-preview-2414--regen-marketplace.netlify.app/project/C02-001

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by blushi** [comment]
@erikalogie see testing instructions

**Comment by erikalogie** [comment]
LGTM!

File: web-marketplace/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { useMemo } from 'react';
```
Comment (r41ph) [inline web-marketplace/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx:1]: It looks like this import isn't being used.

File: web-marketplace/src/pages/Projects/Projects.context.tsx [hunk 2]
```diff
@@ -15,6 +15,7 @@ type ProjectsContextType = {
   prefinanceProjectsCount?: number;
   prefinanceProjectsContent?: Maybe<PrefinanceProjects>;
   pagesCount: number;
+  soldOutProjectsIds: string[];
 };
 
 export const useProjectsContext = (): ProjectsContextType => {
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-marketplace/src/pages/Projects/Projects.context.tsx:21]: it's a specific hook used as outlet context for the /projects child pages, see https://reactrouter.com/en/main/hooks/use-outlet-context#useoutletcontext so I believe that makes sense to differentiate it a bit from regular custom hooks.
This file naming pattern `*.context.tsx` is used for other components too to share context within child routes that  (see Dashboard, MyBridge, etc.)

File: web-marketplace/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx (hunk 1 above)
Comment (blushi) [reply web-marketplace/src/components/organisms/ProjectTopSection/ProjectTopSection.tsx]: removed

**Comment by blushi** [comment]
@r41ph could you have another look at this?

**Review by r41ph — APPROVED** [review]

**Pull request merged**